boolean/boolean.py

```diff
@@ -1175,7 +1175,11 @@ def start_operation(ast, operation):
         elif token == TOKEN_RPAR:
             while True:
                 if ast[0] is None:
-                    raise TypeError('Bad closing parenthesis at line: %(row)d, column: %(col)d' % locals())
+                    if isinstance(position, tuple):
+                        row, col = position
+                        raise TypeError('Bad closing parenthesis at line: %(row)d, column: %(col)d' % locals())
+                    else:
+                        raise TypeError('Bad closing parenthesis at position: %(position)d' % locals())
```

Comment:
Note also that the original purpose of return a position was to eventually return a structured message pointing where in the original stream the error was. Yet, this would require a more structured exception that would store the position, the tokenstr and the original expression (if this was a string)
so for now, short of doing that you could keep it even simpler as: 
`Bad closing parenthesis at position: %(position)r` and used this in all cases whether a tuple or int?